File: EFarm_Application-fullstack/Model/Dto/ProductDto.cs
﻿namespace EFarm.Server.Model.Dto
{
	public class ProductDto
	{
        public int Id { get; set; }
        public string Name { get; set; }
		public string? Description { get; set; }
		public decimal? OldPrice { get; set; }
		public decimal Price { get; set; }
		public string? PictureUri { get; set; }
	}
}

File: Safari-webstore-develop/src/main/java/com/decagon/safariwebstore/dto/ProductDTO.java
package com.decagon.safariwebstore.dto;

import com.decagon.safariwebstore.model.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductDTO {
    private Long id;
    private String name;
    private double price;
    private String description;
    private List<Category> category;
    private List<SubCategory> subCategory;
    private List<Size> sizes;
    private List<Color> colors;
    private List<ProductImage> productImages;

    public ProductDTO(Long id, String name, double price, String description, List<Category> category, List<SubCategory> subCategory) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.description = description;
        this.category = category;
        this.subCategory = subCategory;
    }
}
